Polyhedron/demo/Polyhedron/Plugins/IO/3mf_io_plugin.cpp

```diff
@@ -0,0 +1,331 @@
+    }
+    for(int i=0; i< nb_point_sets; ++i)
+    {
+      Scene_points_with_normal_item* pts_item = new Scene_points_with_normal_item();
+      for(std::size_t j = 0; j < all_points[i].size(); ++j)
+      {
+        pts_item->point_set()->insert(all_points[i][j]);
+      }
+      pts_item->setName(names[i].data());
+      pts_item->invalidateOpenGLBuffers();
+      CGAL::Color c = all_colors[i].front();
+      pts_item->setColor(QColor(c.red(), c.green(), c.blue()));
+      pts_item->setProperty("already_colord", true);
+      result << pts_item;
+      if(add_to_scene)
+        CGAL::Three::Three::scene()->addItem(pts_item);
+    }
+    all_points.clear();
+    names.clear();
+    all_colors.clear();
+    int nb_meshes =
+        CGAL::read_soups_from_3mf(fileinfo.filePath().toUtf8().toStdString(),
+                                  all_points, all_polygons, all_colors, names);
+    if(nb_meshes <0 )
+    {
+      ok = false;
+      std::cerr << "Error in reading of meshes."<<std::endl;
+      return result;
+    }
+    for(int i = 0; i< nb_meshes; ++i)
+    {
+      PolygonRange triangles = all_polygons[i];
+      PointRange points = all_points[i];
+      ColorRange colors = all_colors[i];
+      bool ok = true;
+      if(!PMP::is_polygon_soup_a_polygon_mesh(triangles))
+        ok = PMP::orient_polygon_soup(points, triangles);
+      if(!ok)
+      {
+        std::cerr<<"Object is not orientable. Skipped."<<std::endl;
+      }
+      else
+      {
+        SMesh mesh;
+        PMP::polygon_soup_to_polygon_mesh(points, triangles, mesh);
+        CGAL::Color first = colors.front();
+        bool need_pmap = false;
+        for(auto color : colors)
+        {
+          if (color != first)
+          {
+            need_pmap = true;
+            break;
+          }
+        }
+        if(need_pmap)
+        {
+          //todo: check if possibility to store pid as property or not
+          SMesh::Property_map<face_descriptor,CGAL::Color> fcolor =
+              mesh.add_property_map<face_descriptor,CGAL::Color>("f:color",first).first;
+          for(std::size_t pid = 0; pid < colors.size(); ++pid)
+          {
+            put(fcolor, face_descriptor(pid), colors[pid]);//should work bc mesh is just created and shouldn't have any destroyed face. Not so sure bc of orientation though.
+          }
+        }
+        Scene_surface_mesh_item* sm_item = new Scene_surface_mesh_item(mesh);
+        if(first == CGAL::Color(0,0,0,0))
+          first = CGAL::Color(50,80,120,255);
+        sm_item->setColor(QColor(first.red(), first.green(), first.blue()));
+        sm_item->setProperty("already_colored", true);
+        sm_item->setName(names[i].data());
+        sm_item->invalidateOpenGLBuffers();
+        result << sm_item;
+        if(add_to_scene)
+          CGAL::Three::Three::scene()->addItem(sm_item);
+      }
+    }
+    ok = true;
+    return result;
+  }
+
+
+  bool canSave(const CGAL::Three::Scene_item*) Q_DECL_OVERRIDE {return false;}
```

Comment:
false is surprising provided there is a `save()` function just after.

Reply:
PING

Reply:
It can save, but not through the regular pipeline, so canSave must return false. `save()` is called from somewhere else.